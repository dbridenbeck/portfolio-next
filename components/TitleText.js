import { motion, AnimatePresence } from "framer-motion";

const variants = {
  moveUp: { opacity: 0, y: "0px", x: "-100px" },
  center: { opacity: 1, y: "0", x: "0px" },
  moveRight: { opacity: 0, y: "-100px", x: "0px" },
};

const TitleText = ({ text, onHelloPage }) => (
  <>
    <AnimatePresence>
      {onHelloPage && (
        <motion.div
          className="title"
          initial="moveUp"
          animate="center"
          exit="moveRight"
          variants={variants}
        >
          <h1 className="title">{text}</h1>
        </motion.div>
      )}
    </AnimatePresence>
    <AnimatePresence>
      {!onHelloPage && (
        <motion.div
          className="title" 
          initial="moveUp"
          animate="center"
          exit="moveRight"
          variants={variants}
        >
          <h1 className="title">{text}</h1>
        </motion.div>
      )}
    </AnimatePresence>

    <style jsx>{`
      .title {
        position: absolute;
        font-size: 1em;
        font-weight: 300;
        margin: -5% 0 0 10%;
      }
    `}</style>
  </>
);

export default TitleText;