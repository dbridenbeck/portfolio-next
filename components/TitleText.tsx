import styled from 'styled-components';
import { AnimatePresence } from "framer-motion";
import Head from "next/head";
import TitleAnimation from "../animations/TitleAnimation";

const Title = styled.h1`
  position: absolute;
  margin: -5% 0 0 2.5%;
  padding: 0;
  font-family: "Crimson Text", serif;
  font-size: 4em;
  font-weight: 200;
  line-height: 1em;
  color: ${({ color }) => (color === "red" ? "#d13b40" : "#3bc9d1")}; 
  @media (min-width: 1270px) {
    font-size: 6em;
  }
`;

interface TitleTextProps {
  currentPage: string;
}

const TitleText: React.FC<TitleTextProps> = ({ currentPage }) => {
  return (
    <>
      <Head>
        <link
          href="https://fonts.googleapis.com/css2?family=Crimson+Text&display=swap"
          rel="stylesheet"
        />
      </Head>
      {/* 
        check for currentPage & !currentPage used to add/remove from DOM
        which triggers AnimatePresence to fire animation
      */}
      <AnimatePresence initial={false}>
        {currentPage === "HELLO" && (
          <TitleAnimation>
            <Title color={"blue"}>{currentPage}</Title>
          </TitleAnimation>
        )}
      </AnimatePresence>
      <AnimatePresence initial={false}>
        {currentPage === "ABOUT" && (
          <TitleAnimation>
            <Title color={"red"}>{currentPage}</Title>
          </TitleAnimation>
        )}
      </AnimatePresence>
    </>
  );
};

export default TitleText;
