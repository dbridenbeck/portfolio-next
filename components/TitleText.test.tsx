import React from "react";
import { render } from "@testing-library/react";
import TitleText from "./TitleText";

describe("TitleText", () => {
  it("should render without errors", () => {
    const { container } = render(<TitleText />);
    expect(container).toMatchSnapshot();
  });
});
