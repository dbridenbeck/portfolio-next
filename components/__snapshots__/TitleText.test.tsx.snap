// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleText should render without errors 1`] = `
<div>
  <div
    class="TitleText__TitleContainer-qiiz6j-0 RsITt"
  >
    <h1
      class="TitleText__Title-qiiz6j-1 GdaQb"
    >
      I'm Darren Bridenbeck,
    </h1>
    <h2
      class="TitleText__SubTitle-qiiz6j-2 cJlTbs"
    >
      a software engineer.
    </h2>
  </div>
</div>
`;
